src/main/java/cc/hyperium/mods/discord/DiscordPresence.java

```diff
@@ -0,0 +1,157 @@
+/*
+ *     Copyright (C) 2018  Hyperium <https://hyperium.cc/>
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU Lesser General Public License as published
+ *     by the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU Lesser General Public License for more details.
+ *
+ *     You should have received a copy of the GNU Lesser General Public License
+ *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package cc.hyperium.mods.discord;
+
+import cc.hyperium.Hyperium;
+import cc.hyperium.config.Settings;
+import cc.hyperium.event.EventBus;
+import cc.hyperium.event.GuiOpenEvent;
+import cc.hyperium.event.InvokeEvent;
+import cc.hyperium.event.JoinMinigameEvent;
+import cc.hyperium.event.ServerJoinEvent;
+import cc.hyperium.event.SingleplayerJoinEvent;
+import cc.hyperium.event.minigames.Minigame;
+import cc.hyperium.gui.GuiHyperiumScreenMainMenu;
+import net.arikia.dev.drpc.DiscordEventHandlers;
+import net.arikia.dev.drpc.DiscordRPC;
+import net.arikia.dev.drpc.DiscordRichPresence;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiChat;
+import net.minecraft.client.gui.GuiMultiplayer;
+import net.minecraft.client.gui.GuiSelectWorld;
+
+public class DiscordPresence {
+
+    private long startTime;
+
+    public void load() {
+        if (Settings.DISCORD_RP) {
```

Comment:
In case you didn't realise, this will mean that users have to restart in order to get rich presence to work. And will have to restart in order to disable it too.

Reply:
I believe it is already that way

Reply:
yeah, i've always had to restart in order to deactivate / enable rp, idk how else i'd do it

Reply:
had an idea on how to do it, i'll test it later

Reply:
did you try registering it onworldjoin and unregistering upon world leave? I think that might work